swarm/api/uri.go

```diff
@@ -19,9 +19,17 @@ package api
 import (
 	"fmt"
 	"net/url"
+	"regexp"
 	"strings"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/swarm/storage"
 )
 
+//matches hex swarm hashes
+// TODO: this is bad, it should not be hardcoded how long is a hash
```

Comment:
@nolash has recently merged multihashes into `bzz-resource` scheme. i know that this method is not used by it but this might be quite problematic if we decide to allow fetching content by multihashes with the normal `bzz` scheme. and i'm guessing this is not so far down the road.
@zelig @nolash ?

Reply:
This does not exactly pertain to this PR, as all I did was to move the key parsing logic to `uri.go`, including the TODO comment. (I thought I'd rather have that parsing capability in there rather than in `api.go`, felt to me a bit lost in there)